src/services/annotations-parser.js

```diff
@@ -289,3 +321,31 @@ export async function parseExternalAnnotationResource(annotation) {
     };
   });
 }
+
+/**
+ * Generate a random color for annotation sets compliant with WCAG
+ * 2.0 level AA for normat text
+ * Reference: https://stackoverflow.com/q/43193341/4878529
+ * @returns {String} HSL color code
+ */
+function generateColor(existingColors) {
+  let newColor;
+  const getNewColor = () => {
+    const hue = Math.floor(Math.random() * 360);
+    /**
+     * saturation and lightness are set fixed values to acheive 
+     * WCAG compliant contrast ratio of 4.5 for normal texts
+     */
+    const saturation = 80;
+    const lightness = 90;
+    newColor = `hsl(${hue}, ${saturation}%, ${lightness}%)`;
+  };
+  getNewColor();
+
+  // If the generated color is already used generate another color
+  if (existingColors.length > 0 && existingColors.includes(newColor)) {
+    getNewColor();
+  } else {
+    return newColor;
+  }
```

Comment:
It is actually a while loop constructed with `if/else`.
The purpose of this was to get unique random colors for each of the tags presented in the annotations UI. I'm afraid with a limit on the number of attempts it might not serve its purpose?

Reply:
@cjcolvar I will merge this for now, maybe we can re-visit this later?